robottelo/ui/computeresource.py

```diff
@@ -75,34 +119,29 @@ def search(self, name):
         element = self.search_entity(name, locators["resource.select_name"])
         return element
 
-    def update(self, oldname, newname, orgs, new_orgs, org_select=False,
-               provider_type=None, url=None, user=None, password=None,
-               region=None, libvirt_display=None, libvirt_set_passwd=True,
-               tenant=None, new_description=None):
-        """Updates a compute resource."""
-        element = self.search(oldname)
+    def update(self, name, newname=None, parameter_list=[],
+               orgs=None, org_select=None, locations=None, loc_select=None):
+        """Updates compute resource entity"""
+        element = self.search(name)
```

Comment:
Checking for the element every time search method is called made me thing on improving this and maybe we can improve in the future to search method raise that exception. If this is not the behaviour 100% of the calls, we can add a flag to not raise an exception because I think this is the less common case.